packages/editor/src/hooks/custom-sources-backwards-compatibility.js

```diff
@@ -116,3 +116,19 @@ addFilter(
 	'core/editor/custom-sources-backwards-compatibility/shim-attribute-source',
 	shimAttributeSource
 );
+
+// The above filter will only capture blocks registered after the filter was
+// applied. There may already be registered by this point, and those must be
+// updated to apply the shim.
+//
+// The following implementation achieves this, albeit with a couple caveats:
+// - Only blocks registered on the global store will be modified.
+// - The block settings are directly mutated, since there is currently no
+//   mechanism to update an existing block registration. This is the reason for
+//   `getBlockType` separate from `getBlockTypes`, since the latter returns a
+//   _copy_ of the block registration (i.e. the mutation would not affect the
+//   actual registered block settings).
+globalSelect( 'core/blocks' )
+	.getBlockTypes()
+	.map( ( { name } ) => globalSelect( 'core/blocks' ).getBlockType( name ) )
```

Comment:
Let's add a note to `getBlockType` to make sure that in the future, someone doesn't change it to return a copy.

Or, better yet, let's make an experimental action to update the registration.

Reply:
I thought about this as well, but figured the end-to-end test would be enough to protect ourselves from regression (it would start to fail if `getBlockType` returns a copy).

I do think a function to update a block registration would be a good idea, since it seems like a valid use-case. Needing to resort to mutating the settings object is definitely not ideal. I can create an issue to track this.

Reply:
:+1:

Reply:
Issue: #20548